src/database/queries/applicationFormFields/selectByProposalId.sql

```diff
@@ -10,4 +10,4 @@ INNER JOIN proposal_field_values pfv
 INNER JOIN proposal_versions pv
   ON pv.id = pfv.proposal_version_id
 WHERE pv.proposal_id = :proposalId
-ORDER BY pv.version DESC, pfv.position;
+ORDER BY pv.version DESC, pfv.position, pfv.id;
```

Comment:
While I agree that doing application joins isn't the right approach (and we have a long term plan to move from it) I'll also throw out that so long as we are, a cleaner solution, IMHO, is to implement a joining algorithm that does not require artificially coupled queries -- e.g. one that will take data and, if it needs sorting, just sort it rather than throwing an error.

From a design PoV I think that is more maintainable than this approach, as this now has two unrelated queries that need to be sorted in a specific way for reasons that don't actually make sense in the context of the queries themselves.

I assume the desire to not sort is an optimization decision, but we don't currently have reason to believe that the optimization is necessary.  On the flip side, the optimization has resulted in this bug and if we merge the PR I would argue it is resulting in tech debt in the form of unintuitive coupling.

Reply:
> …if we merge the PR I would argue it is resulting in tech debt in the form of un-intuitive coupling.

If this design is so wrong that we should prioritize a wider fix, that's reasonable, but I don't think it should block pushing out this production hotfix.

Not to be demo-driven, but we _do_ have a big demo tomorrow and likely a bunch of stakeholders logging in right after, and their experience with a broken site is net-worse for the project than this relatively benign change, which is at least idiomatic to our approach even if we're operating under the wrong idiom.

Reply:
Wait wait, I have a better reply:

PULL REQUESTS WELCOME

:trollface:

Reply:
Well to be clear I'm not proposing a wider fix, just that instead of sorting in the query we should just invoke .sort((a,b) => ...) in the immediate context of where a particular context-specific order matters.

(also re the PR joke -- I don't think we should get in the habit of reviewers needing to take on the burden of owning an issue simply because they have feedback!)

It's fine to decide to merge tech debt and open an issue to capture that tech debt!

Reply:
(I wanna triple down on the last part though: I intentionally left this as a *comment* rather than "request changes" in the name of not blocking progress!)

Reply:
Fully agree that reviewers _have_ to be free to suggest improvements without a requisite burden of providing them, or else they'll hold their feedback and that's bad! My joke was truly a non sequitur and I apologize for the insinuation. While it's obviously not up to me, diving into a PR that provides the fix wouldn't be how I would prioritize your time, that's how much of a joke it was.

Re: the approach, I'll bow out and let you guys discuss it further, but by idiomatic here I meant "we're already sorting in the query; fix that sort to not blow up" and by non-idiomatic I mean "changing where we sort".

Reply:
So clearly "idiomatic" is the wrong word here. "Minimally different from current design but not broken" perhaps?

Reply:
FWIW I was thinking of the current sorting in the query as being use-case-agnostic (e.g. selecting something by proposal ID can happen for all sorts of reasons, we picked a way to sort that makes generic sense).  Whereas this new sort is very specific to JUST this use case of selecting in order to join proposal field values.

That said, I now get what you're saying, which I think is essentially "it may be that it's inappropriate to rely on the query to sort, but we were already doing that, and we might as well update the arbitrary sort order to fix the situation even if we shouldn't be relying on arbitrary sort order"

So, FINE JUSTIN YOU ARE RIGHT (even if I think the most correct fix to overly coupled pieces of code is to decouple them gosh darn it!)

Reply:
> FWIW … ID … JUST … FINE JUSTIN YOU ARE RIGHT

My speed-reading strategy of skimming for all-caps words continues to pay off!